hugolib/site.go

```diff
@@ -1800,16 +1800,18 @@ func (s *Site) RenderHomePage() error {
 		}
 	}
 
-	// TODO(bep) reusing the Home Node smells trouble
-	n.URL = helpers.URLize("404.html")
-	n.IsHome = false
-	n.Title = "404 Page not found"
-	n.Permalink = s.permalink("404.html")
-	n.scratch = newScratch()
-
-	nfLayouts := []string{"404.html"}
-	if nfErr := s.renderAndWritePage("404 page", "404.html", n, s.appendThemeTemplates(nfLayouts)...); nfErr != nil {
-		return nfErr
+	if !viper.GetBool("Disable404") {
```

Comment:
It's much easier to read if you do if viper.GetBool("Disable404") return nil instead.